02_Catalog_Queries_with_TAP.ipynb

```diff
@@ -0,0 +1,1358 @@
+    "\n",
+    "This search usually takes about 2-10 seconds."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "query = \"SELECT objectId, detect_isPrimary, \" + \\\n",
+    "        \"coord_ra AS ra, coord_dec AS dec, \" + \\\n",
+    "        \"scisql_nanojanskyToAbMag(g_cModelFlux) AS mag_g_cModel, \" + \\\n",
+    "        \"scisql_nanojanskyToAbMag(r_cModelFlux) AS mag_r_cModel, \" + \\\n",
+    "        \"scisql_nanojanskyToAbMag(i_cModelFlux) AS mag_i_cModel, \" + \\\n",
+    "        \"r_extendedness \" + \\\n",
+    "        \"FROM dp02_dc2_catalogs.Object \" + \\\n",
+    "        \"WHERE CONTAINS(POINT('ICRS', coord_ra, coord_dec), \" + \\\n",
+    "        \"CIRCLE('ICRS', \" + use_center_coords + \", \" + use_radius + \")) = 1 \" + \\\n",
+    "        \"AND detect_isPrimary = 1 \" + \\\n",
+    "        \"AND scisql_nanojanskyToAbMag(g_cModelFlux) > 17.0 \" + \\\n",
+    "        \"AND scisql_nanojanskyToAbMag(g_cModelFlux) < 23.0 \" + \\\n",
+    "        \"AND scisql_nanojanskyToAbMag(r_cModelFlux) > 17.0 \" + \\\n",
+    "        \"AND scisql_nanojanskyToAbMag(r_cModelFlux) < 23.0 \" + \\\n",
+    "        \"AND scisql_nanojanskyToAbMag(i_cModelFlux) > 17.0 \" + \\\n",
+    "        \"AND scisql_nanojanskyToAbMag(i_cModelFlux) < 23.0 \" + \\\n",
+    "        \"AND r_extendedness IS NOT NULL \"\n",
+    "results = service.search(query).to_table().to_pandas()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Display the Pandas table."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4.2. Data preparation\n",
+    "\n",
+    "The basis for any data visualization is the underlying data.\n",
+    "We will prepare ColumnDataSource (CDS) from the data returned by the query above that can be passed directly to bokeh.\n",
+    "The CDS is the core of bokeh plots.\n",
+    "Bokeh automatically creates a CDS from data passed as python lists or numpy arrays.\n",
+    "CDS are useful as they allow data to be shared between multiple plots and renderers, enabling brushing and linking.\n",
+    "A CDS is essentially a collection of sequences of data that have their own unique column name. \n",
+    "\n",
+    "Getting the data preparation phase right is key to creating powerful visualizations. \n",
+    "\n",
+    "Recall that we had defined: `use_center_coords = \"62, -37\"`. Get the central RA and Dec as values (datatype float)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "temp = use_center_coords.split(', ')\n",
+    "center_ra = float(temp[0])\n",
+    "center_dec = float(temp[1])\n",
+    "print(center_ra, center_dec)"
+   ]
```

Comment:
This is not necessary if you use an astropy SkyCoord object - you can avoid sting splits and temp objects.

Reply:
fixed